test_data/sdk_compiled_pipelines/valid/critical/pipeline_with_workspace.py

```diff
@@ -53,6 +53,7 @@ def read_from_workspace(file_path: str) -> str:
     name="pipeline-with-workspace",
     description="A pipeline that demonstrates workspace functionality",
     pipeline_config=dsl.PipelineConfig(
+        pipeline_run_parallelism=None,
```

Comment:
Even this working as a test for no parallelism, I think we should create a specific pipeline to test this particular scenario. My concern is that this pipeline may change or be removed in the future for whatever reason and we may not notice that this also tests no parallelism (beside workspace). We would end up with no tests for no parallelism.